nx/lib/nx.ex

```diff
@@ -5244,6 +5244,82 @@ defmodule Nx do
     impl!(tensor).conjugate(%{tensor | type: Nx.Type.to_complex(tensor.type)}, tensor)
   end
 
+  @doc """
+  Returns the real component of each entry in a complex tensor
+  as a floating point tensor.
+
+  ## Examples
+
+      iex> Nx.real(Complex.new(1, 2))
+      #Nx.Tensor<
+        f32
+        1.0
+      >
+
+      iex> Nx.real(Nx.tensor(1))
+      #Nx.Tensor<
+        f32
+        1.0
+      >
+
+      iex> Nx.real(Nx.tensor([Complex.new(1, 2), Complex.new(2, -4)]))
+      #Nx.Tensor<
+        f32[2]
+        [1.0, 2.0]
+      >
+  """
+  @doc type: :element
+  def real(tensor) do
+    tensor = to_tensor(tensor)
+
+    type =
+      case tensor.type do
+        {:f, _} = type -> type
+        {:c, size} -> {:f, div(size, 2)}
+        _ -> {:f, 32}
+      end
+
+    impl!(tensor).real(%{tensor | type: type}, tensor)
```

Comment:
I am thinking we should specialize this. If the tensor is floating, then we don't need to call the backend at all. If the tensor is complex, then we call the backend. Otherwise it is the equivalent of casting to float (we can use `as_type`).